demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+         * 
+         * @param responses The 'responses' variable can be any data type you would like it to be depending on the format of your custom question's responses. 
+         * For instance, it might be a string for the value of a simple text input
+         * Or for a quesiton asking a student to plot coorinates on a graph, it might be an object containing the x and y coordinates, e.g.
+         *     {
+         *       x: 123,
+         *       y: 408
+         *     }
+         */
+
+        /** EXAMPLE implementation
+         * 
+         * document.getElementById('my-radio-button').addEventListener('click', (event) => {
+         *   
+         *  // STEP 1: when the user clicks on your custom question's input, save its current value as the value of responses
+         *  let responses = event.target.value
+         *   
+         *  // STEP 2: 
+         *  // now notfy Questions API that the user has changed their response
+         *   
+         *   events.trigger('changed', responses);
+         * 
+         *  // STEP 3 - OPTIONAL
+         *  // If you have chosen to implement facade.showValidationUI() above to display visiual feedback for a correct / incorrect answer
+         *     when the user presses "Check Answer" for the current response:
+         *     then make sure you have also implemented facade.resetValidationUI(), and that you call it here:
+         * 
+         * // facade.resetValidationUI()
+         * 
+         * })
+         * 
+         */
+
+        let response;
+        const shapeButtons = Array.from(el.querySelectorAll('.shape-wrapper button'));
+        shapeButtons.forEach((shapeButton, index) => {
+            shapeButton.addEventListener('click', () => {
+                shapeButtons.forEach(btn => {
+                    if(btn.classList.contains('selected')) {
+                        btn.classList.remove('selected')
+                    }
+                })
+                shapeButton.classList.toggle('selected');
+                shapeButton.parentElement.style.outlineColor = question.selected_shape_color;
+                
+                response = shapeButton.getAttribute('data-index');
+                events.trigger('changed', response);
+                facade.resetValidationUI();
+            })
+        })
+
+
+        // "validate" event can be triggered when Check Answer button is clicked or when public method .validate() is called
+        // so developer needs to listen to this event to decide if he wants to display the correct answers to user or not
+        // options.showCorrectAnswers will tell if correct answers for this question should be display or not.
+        // The value showCorrectAnswers by default is the value of showCorrectAnswers inside initOptions object that is used
+        // to initialize question app or the value of the options that is passed into public method validate (like question.validate({showCorrectAnswers: false}))
```

Comment:
Fix the comments for readability
```suggestion
        /**
          * "validate" event can be triggered when Check Answer button is clicked or when
          * public method .validate() is called so developer needs to listen to this event to decide
          * if he wants to display the correct answers to user or not
          * 
          * options.showCorrectAnswers will tell if correct answers for this question should be display or not.
          * The value showCorrectAnswers by default is the value of showCorrectAnswers
          * inside initOptions object that is used
          * 
          *
          * to initialize question app or the value of the options that is passed into public method validate
          *  (like question.validate({showCorrectAnswers: false}))
          */
```